R/predict_utils.R

```diff
@@ -83,6 +83,10 @@ ready_no_prep <- function(training_data, newdata) {
 #' @noRd
 ready_with_prep <- function(object, newdata, mi = extract_model_info(object)) {
   recipe <- attr(object, "recipe")
+  # If dummies, convert all characters to factors to set reference
+  if (grepl("Dummy variables from", recipe))
```

Comment:
`recipe` is a big nested list here, right? It's not clear to me what's being checked against. Admittedly, it's unlikely that "Dummy variables from" would come from anything else, but be more targeted with your checks to avoid bugs. It will also make your code more readable and make it easier to find, e.g. the object with the `recipe` here that has to do with reference levels.